In [1]:
#Importing Libraries
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [3]:
df = pd.read_csv('/content/gdrive/MyDrive/Files/income.csv')
df

Name  Monthly Income ($)
0        Rob                5000
1      Rafiq                6000
2       Nina                4000
3      Sofia                7500
4      Mohan                8000
5        Tao                7000
6  Elon Musk            10000000

* We have the following Columns:
  1. Name
  2. Monthly Income

* We will rename the columns for convinence

In [8]:
df = df.rename(columns={'Name':'name','Monthly Income ($)':'income'})
df

name    income
0        Rob      5000
1      Rafiq      6000
2       Nina      4000
3      Sofia      7500
4      Mohan      8000
5        Tao      7000
6  Elon Musk  10000000

In [9]:
df.describe()

income
count  7.000000e+00
mean   1.433929e+06
std    3.777283e+06
min    4.000000e+03
25%    5.500000e+03
50%    7.000000e+03
75%    7.750000e+03
max    1.000000e+07

<b>We have described the following data </b>

* Count: Total no of elements in the csv

* mean: Sum of all elements/total number of elements

<b> Quantile Range </b>

* 25% of the data
* 50% of the data
* 75% of the data
* <b>25% - 75% of the data is known as Inter-Quantile Range </b>

In the above data,

* 25% of the data lies at 5.5
* 50% of the data lies at 7.0
* 75% of the data lies at 7.75


In [13]:
#Describing the quantile range
print(f"25%: {df.income.quantile(0.25)}\n 50%: {df.income.quantile(0.50)}\n 75%: {df.income.quantile(0.75)}")

25%: 5500.0
 50%: 7000.0
 75%: 7750.0


In [17]:
#Couple of ways to interpret quantiles
print(f"25% Lower:",df.income.quantile(0.25,interpolation="lower"))
print(f"25% Higer:",df.income.quantile(0.25,interpolation="higher"))

25% Lower: 5000
25% Higer: 6000


5000 (Lower) ──── 5500 (Linear) ──── 6000 (Higher)


In [18]:
#0 Quantile
df.income.quantile(0)

4000.0

In [19]:
#1 Quantile / Highest Quantile
df.income.quantile(1)

10000000.0

In [20]:
# Removing the outlier
# Before we reomve the outlier, we will understand where the outlier is present
# By viewing the data we can suggest, 6th row is an outlier

In [24]:
identify_the_value = df.income.quantile(0.90)
identify_the_value

4004800.0000000037

In [26]:
#We can see at quantile range 90%, it has higher income, an noise!
df = df[df.income<identify_the_value]
df

name  income
0    Rob    5000
1  Rafiq    6000
2   Nina    4000
3  Sofia    7500
4  Mohan    8000
5    Tao    7000

* We have successfully pre-processed the data by removing the outlier, which now could be used for analytical purpose